### DISCUSSION TITLE
Can I use FreeRDP to control windows equipment connected to KVM device and control from Web (Angular & dotNet) project?

### DISCUSSION THREAD

**Discussion by Rashedujjaman** [FreeRDP/FreeRDP#10660] [Q&A] [answered]
Hi there, I am in a new project of remote equipment management system. I want to control equipments remotely through website that is created on angular and dotNet core technology. Each host will be connected to KVM device that comes with build in vnc. 
Objective is frontend side user will select an equipment and click connect and he can view the equipment screen and control.

To achieve this I will pass equipment ip from frontend to backend and backend will call FreeRDP and pass host's ip, password, username or anything else that is required to connect remote vnc. If the connection is successfull it should return the equipment's screen along with the control access.

Is this possible with FreeRDP?

I am trying to configure FreeRDP since 3 days and at last today I successfully configured FreeRDP but can't build. I need someone to help guiding the building.

Here is the error message that preventing to build: 
c:\msys64>cmake --build freerdp-build
[0/2] Re-checking globbed directories...
[3/384] Building C object winpr/libwinpr/CMakeFiles/winpr.dir/wtsapi/wtsapi_win32.c.obj
FAILED: winpr/libwinpr/CMakeFiles/winpr.dir/wtsapi/wtsapi_win32.c.obj
C:\msys64\mingw64\bin\cc.exe -DFREERDP_EXPORTS -DNDEBUG -DUNICODE -DWIN32_LEAN_AND_MEAN -DWINPR_EXPORTS -DWINPR_TIMEZONE_FILE=\"/var/db/zoneinfo\" -DWINVER=0x0601 -DWITH_JSONC -DWITH_OPENSSL -DWITH_SMARTCARD_EMULATE -DWITH_SMARTCARD_PCSC -DWITH_TIMEZONE_COMPILED -DWITH_URIPARSER -DWITH_VERBOSE_WINPR_ASSERT -D_CRT_SECURE_NO_WARNINGS -D_UNICODE -D_WIN32_WINNT=0x0601 -D_WINSOCK_DEPRECATED_NO_WARNINGS -IC:/msys64/freerdp/winpr/include -IC:/msys64/freerdp-build/winpr/include -IC:/msys64/freerdp/winpr/libwinpr/utils -IC:/msys64/mingw64/include/json-c -IC:/msys64/freerdp/winpr/libwinpr/utils/. -Wall -Wpedantic -Wno-padded -Wno-switch-enum -Wno-cast-align -Wno-declaration-after-statement -Wall -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls -fno-omit-frame-pointer -O3 -DNDEBUG -std=gnu11 -flto=auto -fno-fat-lto-objects -MD -MT winpr/libwinpr/CMakeFiles/winpr.dir/wtsapi/wtsapi_win32.c.obj -MF winpr\libwinpr\CMakeFiles\winpr.dir\wtsapi\wtsapi_win32.c.obj.d -o winpr/libwinpr/CMakeFiles/winpr.dir/wtsapi/wtsapi_win32.c.obj -c C:/msys64/freerdp/winpr/libwinpr/wtsapi/wtsapi_win32.c

**Comment by akallabeth** [comment, marked as answer]
@Rashedujjaman `FreeRDP` implements the [RDP](https://en.wikipedia.org/wiki/Remote_Desktop_Protocol) protocol commonly found in microsoft windows.
[VNC](https://en.wikipedia.org/wiki/VNC) is not supported at all by this project.

[edit] as for compilation, check https://github.com/FreeRDP/FreeRDP/wiki/Compilation#compiling-for-mingw for details

**Reply by Rashedujjaman** [reply]
@akallabeth Thank you so much. 

Are you familiar with any other solution that can meet my requirements?
